### PR TITLE
Adding support for global SmartStore in hybrid

### PR DESCRIPTION
**Pull request by bhariharan**
* Moved ```STORE_CURSORS``` to the ```SmartStore``` class, so that it is per ```SmartStore``` instance.
* Added a ```boolean``` argument that can be passed in for a global database. The default is still the user's database.

### CODE REVIEW

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/phonegap/SmartStorePlugin.java [hunk 1]
```diff
@@ -54,35 +53,34 @@
...
+	private static final String SOUP_NAME = "soupName";
+	private static final String PATH = "path";
+	private static final String PATHS = "paths";
+	private static final String QUERY_SPEC = "querySpec";
+	private static final String EXTERNAL_ID_PATH = "externalIdPath";
+	private static final String ENTRIES = "entries";
+	private static final String ENTRY_IDS = "entryIds";
+	private static final String INDEX = "index";
+	private static final String INDEXES = "indexes";
+	private static final String IS_GLOBAL_STORE = "isGlobalStore";
```
Comment (bhariharan) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/phonegap/SmartStorePlugin.java]: New key for global.

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/phonegap/SmartStorePlugin.java [hunk 2]
```diff
@@ -165,18 +165,20 @@ public void run() {
...
 		Long[] soupEntryIds = new Long[jsonSoupEntryIds.length()];
-		for (int i=0; i<jsonSoupEntryIds.length(); i++)
-			soupEntryIds[i] = jsonSoupEntryIds.getLong(i); 
+		for (int i = 0; i < jsonSoupEntryIds.length(); i++) {
+			soupEntryIds[i] = jsonSoupEntryIds.getLong(i);
+		}
 		
 		// Run remove
-		SmartStore smartStore = getSmartStore();
+		final SmartStore smartStore = (isGlobal ? getGlobalSmartStore() : getUserSmartStore());
```
Comment (bhariharan) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/phonegap/SmartStorePlugin.java]: This is all that's new everywhere.

**Comment by wmathurin** [comment]
Looks good.

**Comment by bhariharan** [comment]
Moved the cursor map back to the plugin in [this](https://github.com/bhariharan/SalesforceMobileSDK-Android/commit/cbc3d5ef964e285d1f2be019dfb8f6d1a0bf6e64) commit. JS commits coming up for ```shared``` repo soon.

**Comment by khawkins** [comment]
Looks okay to me.

**Pull request merged**